### PR TITLE
Support Op conversion from mhlo::ScatterOp to lmhlo::ScatterOp

### PR DESCRIPTION
**Pull request by eedalong**

### CODE REVIEW

File: tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc [hunk 1]
```diff
@@ -47,6 +47,8 @@ limitations under the License.
 #include "mlir/Pass/Pass.h"
 #include "mlir/Transforms/DialectConversion.h"
 
+#include <iostream>
```
Comment (Yancey0623) [inline tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc]: remove the unused header file.

File: tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc [hunk 2]
```diff
@@ -348,6 +348,69 @@ struct HloToLhloReturnOpConverter : public BaseOpConversion<mhlo::ReturnOp> {
...
+    auto& entryBlock = newOp.getUpdateComputation().front();
+    TypeConverter::SignatureConversion sigConversion(
+        adaptor.getOperands().size());
+    for (auto arg : entryBlock.getArguments()) {
+      auto oldType = arg.getType().template cast<TensorType>();
+      auto newType =
+          MemRefType::get(oldType.getShape(), oldType.getElementType());
+      sigConversion.addInputs(arg.getArgNumber(), newType);
+    }
+    auto returnOp = cast<mhlo::ReturnOp>(entryBlock.getTerminator());
```
Comment (Yancey0623) [inline tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc:386]: I'm not sure the tuple result type is necessary, do you want to add a ut to test this case?

File: tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc (hunk 1 above)
Comment (eedalong) [reply tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc]: Fixed.

File: tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc (hunk 2 above)
Comment (eedalong) [reply tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc:386]: Fixed. Removed logic about tuple result processing, reduce body in scatterop only return one result, not tuple.

**Review by Yancey0623 — APPROVED** [review]
LGTM!

**Pull request merged**